### ISSUE TITLE
[7.x] Route URL generator gives "Missing Required Parameters" error for CACHED routes with custom keyed parameters.

### ISSUE THREAD

**Issue by AramZahedi** [laravel/framework#31744]
- Laravel Version: 7.0.3
- PHP Version: 7.4.3

### Description:
After caching routes, URL generator cannot recognize given route parameters for routes that have Custom Keyed Parameters.
Everything works fine if routes have not been cached, but once cached, you'll get **UrlGenerationException** error after using `route('your-route')` for any route with custom keyed parameters, stating that required parameters for the route are missing.

### Steps To Reproduce:
1. Add this block of code to your web.php routes file:
```php
Route::prefix('profile/{user:username}/portfolios')->group(function () {
    /*
     |===============================================
     |                Store Portfolio
     |===============================================
     */
    Route::post('/', 'HomeController@index')
        ->name('profile.portfolios.store');
});
```

2. Cache the routes in the command line using:
`php artisan route:cache`

3. Run this line of code in a phpunit test:
```php
echo route('profile.portfolios.store', [
    'user' => 111
]);
```
And you'll get this error for sure:

> Illuminate\Routing\Exceptions\UrlGenerationException : Missing required parameters for [Route: profile.portfolios.store] [URI: profile/{user:username}/portfolios/store]

### Where the Problem Comes From:
**THIS PROBLEM ONLY APPEARS WHEN ROUTES ARE CACHED.**
What causes this problem after routes are cached, is the Custom Keyed Parameter set for the route.
The `route('your-route')` function works perfectly if you call it as follows:
```php
echo route('profile.portfolios.store', [
    'user:username' => 111
]);
```
**Output:**

> http://localhost/profile/111/portfolios/store

**Comment by driesvints** [comment]
Sent in a PR here: https://github.com/laravel/framework/pull/31748

**Comment by AramZahedi** [comment]
> Sent in a PR here: #31748

@driesvints Seems like your PR has not been accepted.

**Comment by GrahamCampbell** [comment]
A fix is still being worked on, don't worry. :)

**Comment by taylorotwell** [comment]
Will be fixed in 7.0.4.

**Comment by taylorotwell** [comment]
7.0.4 has been released to fix this.
